### PR TITLE
Parameter Refactor

### LINKED ISSUE #28: Resolution is wrong when using binning / decimation.
First of all, what is the difference between binning and decimation ?

I believe `decimation_x` and `binning_x` will reduce the width. And I believe `decimation_y` and `binning_y` will reduce the height. Am I right ? But 

When setting the launch parameters : 
 -  `binning_x` and `binning_y` to 2
 OR
 -  `decimation_x` and `decimation_y` to 2

I believe that the values of the parameters  `width` and `height` should be divided by `2` as well. Because those values will be outputted in the `camera_info` topic. am I right ?

It works fine for the y values (height) but doesn't for the x values (width). Setting the width to half will crop the image half just like if the decimation happened after the cropping which shouldn't be the case.

---

Config that works :
 -  `binning_x` and `binning_y` to 2 OR `decimation_x` and `decimation_y` to 2
- Height divided by 2 (`1032`)
- Width unchanged (`1544`)

In the topic `/camera_fx/image_raw` we get images of good resolution : `1032x772`

:warning: But the reported resolution in `/camera_fx/camera_info` is wrong, giving the values given as parameters : `1032x1544`.

---

Have you ever had this problem ?

Thanks !

### LINKED ISSUE #37: Review all ROS params, dynamic reconfigure, and launch files.
Make sure all parameters use the same names in both dyn reconfigure file and launch params.

For example, `x_offset` should be `roi_offset_x`

### PR DESCRIPTION
**Pull request by icolwell-as**
Resolves #37 by reviewing all parameters and fixing/updating as-needed. 
I decided to remove the `roi_*` parameters in favour of the feature names the camera uses: `width, height, offset_x, offset_y`
I think this will make it much easier for people to configure the camera as they desire, since the parameter names match what they see in VimbaViewer. The roi details of of the published camera_info topic will be generated according to the vimba settings instead of the other way around.

I believe this also resolves #28 as there were some existing bugs where the camera_info topic used to just assume the ros params are correct, even if the camera can't use the ros params (some camera models don't support certain features).

I intend to use the new `configureFeature` function for all features in a later PR.

Note that the `AvtVimbaCamera::setFeatureValue` function was moved to a different location in the file, I didn't rewrite it.

### CODE REVIEW

File: include/avt_vimba_camera/avt_vimba_camera.h [hunk 1]
```diff
@@ -79,7 +79,7 @@ class AvtVimbaCamera {
  public:
   AvtVimbaCamera();
   AvtVimbaCamera(std::string name);
-  void start(std::string ip_str, std::string guid_str, bool debug_prints = true);
+  void start(std::string ip_str, std::string guid_str, std::string frame_id, bool debug_prints = true);
```
Comment (jilinzhouas) [inline include/avt_vimba_camera/avt_vimba_camera.h]: use const std::string& instead?

File: include/avt_vimba_camera/avt_vimba_camera.h [hunk 2]
```diff
@@ -79,7 +79,7 @@ class AvtVimbaCamera {
  public:
   AvtVimbaCamera();
   AvtVimbaCamera(std::string name);
```
Comment (jilinzhouas) [inline include/avt_vimba_camera/avt_vimba_camera.h]: the same here: const std::string&

File: launch/Mako_G-319.launch [hunk 3]
```diff
@@ -18,6 +18,6 @@
     <arg name="pixel_format" value="BayerRG8"/>
     <arg name="width" value="2064"/>
     <arg name="height" value="1544"/>
-    <arg name="stream_bytes_per_second" value="119824282"/>
+    <arg name="stream_bytes_per_second" value="124000000"/>
```
Comment (dvangeyn-as) [inline launch/Mako_G-319.launch:21]: Why the change from a seemingly random number to 1 million bytes/sec less than the theoretical max of GigE? Should it be 125 * 10^6?

File: launch/Mako_G-319.launch [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <launch>
   <arg name="name" default="camera" doc="The name of the camera"/>
-  <arg name="frame_id" default="camera" doc="The frame id of the camera"/>
+  <arg name="frame_id" default="$(arg name)" doc="The frame id of the camera"/>
   <arg name="ip" default="" doc="The IP for the camera to connect to"/>
   <arg name="guid" default="" doc="The GUID for the camera to connect to"/>
   <arg name="camera_info_url" default="file://$(find avt_vimba_camera)/calibrations/calibration_50-0503343289.yaml"/>
```
Comment (dvangeyn-as) [inline launch/Mako_G-319.launch]: What is the meaning of "50-0503343289" here and the find command in stereo_camera.launch?

File: launch/Mako_G-319.launch (hunk 3 above)
Comment (icolwell-as) [reply launch/Mako_G-319.launch:21]: 124M is the maximum for the Mako G-319 according to their docs. See here page 51:
https://cdn.alliedvision.com/fileadmin/content/documents/products/cameras/Mako/techman/Mako_TechMan_en.pdf
Keep in mind this launch file is specific to the Mako G-319 model.

File: launch/Mako_G-319.launch (hunk 4 above)
Comment (icolwell-as) [reply launch/Mako_G-319.launch]: So this is the calibration file for some camera with a GUID of `50-0503343289` that the original author of the repo had.
The calibration file contains the intrinsic calibration which gets published along with the image for rectification or un-distortion purposes.
Good point though, I'm going to rename the file to "example" or something to reduce confusion.

**Review by jilinzhouas — APPROVED** [review]

**Review by dvangeyn-as — APPROVED** [review]

**Pull request merged**